vignettes/RNAseq_Processing_Workflow.Rmd

```diff
@@ -0,0 +1,283 @@
+
+```{r}
+# create RNA filter object
+myrnafilt <- RNA_filter(omicsData = myseq)
+
+# filter by library size
+plot(myrnafilt, plot_type = "library")
+plot(myrnafilt, plot_type = "library", size_library = 1000000)
+
+```
+
+Next, we utilize the filter to identify any samples with fewer than 5,000 non-zero counts, of which there are none.
+
+```{r}
+# filter based on number or proportion of non-zero counts
+plot(myrnafilt, plot_type = "biomolecule", min_nonzero = 5000)
+```
+
+
+
+
+### Summarize Data
+
+After filtering out any transcripts and/or samples above, we may wish to do some exploratory data analysis.
+
+#### Correlation Heatmap
+
+A Spearman correlation heatmap shows a couple of samples with lower correlation than the rest, but neither of these are 
+
+- Strain B 3 D
+
+- Strain C 1 E
+
+```{r}
+mycor <- cor_result(omicsData = myseq)
+plot(mycor, interactive = TRUE)
+```
+
+
+#### Principal Components Analysis
+
+A principal components analysis (PCA) plot of our data shows clustering of strains A and C with one another, separated from strain B. The two samples that are separated on PC1 one from the rest are the same two samples that stood out in the correlation heatmap.
+
+```{r}
+mypca <- dim_reduction(omicsData = myseq, k=2)
+plot(mypca, interactive = TRUE)
+
+mypca_df <- data.frame(SampleID = mypca$SampleID, PC1 = mypca$PC1, PC2 = mypca$PC2)
+```
+
+
+## Normalization & Statistical Comparisons
+
+For RNAseq data there are three methods available via `pmartR` for making statistical comparisons:
+
+- edgeR [@edger]
+
+- DESeq2 [@deseq2]
+
+- limma-voom [@limmavoom]
+
+Here we utilize all three for demonstration purposes, but in practice a single method should be utilized, the selection of which can be guided by visual inspection of dispersion plots for each method via `dispersion_est`. After using the `diffexp_seq` function to perform statistical comparisons with the selected method, `plot` and `summary` methods are available to help display the results, and the data frame returned by the `diffexp_seq` function can be saved as an Excel file or another useful format if helpful.
+
+
+### edgeR Method
+
+We examine the dispersion plot for use of the edgeR method. 
+
+```{r}
+dispersion_est(omicsData = myseq, method = "edgeR")
+```
+
+
+If we are satisfied with using this method for our statistical comparisons between virus strains, we can do so as follows.
+
+```{r}
+stats_edger <- diffexp_seq(omicsData = myseq, method = "edgeR")
+```
+Various plot types are available to display the results of the statistical comparisons. 
+
+- WHAT ARE THE plot_method SYNTAX options here? Not seeing those in the documentation, but maybe it's on Rachel's latest round of updates? In her notes, she said "Plot of stat results - option for either volcano or MA plot, heat maps with option to subset to most significant molecules (based on p-value)"
+
+```{r warning=FALSE}
+plot(stats_edger, plot_type = "volcano")
```

Comment:
vignette build crashes here due (I think) to a bug in `pmartR::prep_flags`. 
 https://github.com/pmartR/pmartR/blob/241ebe49b6e9afbe96c633629f597d50493e2320/R/plot_fns.R#L5962

It checks for a test type of "EdgeR_LRT", but the statres object has type "EdgeR_F" and so it goes to the wrong logical block and causes problems.  The change was introduced in c1b55e89356a84af67ababf81f40831563e82cde

Reply:
fixed, good catch!